src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/AdvancedBundlingCallbackHandler.java

```diff
@@ -0,0 +1,105 @@
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.field.IdValue;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.em.Bundle;
+import uk.gov.hmcts.reform.iacaseapi.domain.handlers.PreSubmitCallbackHandler;
+import uk.gov.hmcts.reform.iacaseapi.infrastructure.clients.BundleRequestExecutor;
+
+
+@Component
+public class AdvancedBundlingCallbackHandler implements PreSubmitCallbackHandler<AsylumCase> {
+
+    private final BundleRequestExecutor bundleRequestExecutor;
+    private final String emBundlerUrl;
+    private final String emBundlerStitchUri;
+
+    public AdvancedBundlingCallbackHandler(
+        @Value("${emBundler.url}") String emBundlerUrl,
+        @Value("${emBundler.stitch.uri}") String emBundlerStitchUri,
+        BundleRequestExecutor bundleRequestExecutor
+    ) {
+        this.emBundlerUrl = emBundlerUrl;
+        this.emBundlerStitchUri = emBundlerStitchUri;
+        this.bundleRequestExecutor = bundleRequestExecutor;
+
+    }
+
+    public boolean canHandle(
+        PreSubmitCallbackStage callbackStage,
+        Callback<AsylumCase> callback
+    ) {
+        requireNonNull(callbackStage, "callbackStage must not be null");
+        requireNonNull(callback, "callback must not be null");
+
+        return callbackStage == PreSubmitCallbackStage.ABOUT_TO_SUBMIT
+               && callback.getEvent() == Event.GENERATE_HEARING_BUNDLE;
+    }
+
+    public PreSubmitCallbackResponse<AsylumCase> handle(
+        PreSubmitCallbackStage callbackStage,
+        Callback<AsylumCase> callback
+    ) {
+        if (!canHandle(callbackStage, callback)) {
+            throw new IllegalStateException("Cannot handle callback");
+        }
+
+        final AsylumCase asylumCase =
+            callback
+                .getCaseDetails()
+                .getCaseData();
+
+        asylumCase.clear(AsylumCaseFieldDefinition.CASE_BUNDLES);
+        asylumCase.write(AsylumCaseFieldDefinition.BUNDLE_CONFIGURATION, "iac-hearing-bundle-config.yaml");
+        asylumCase.write(AsylumCaseFieldDefinition.BUNDLE_FILE_NAME_PREFIX, getBundlePrefix(asylumCase));
+
+        final PreSubmitCallbackResponse<AsylumCase> response = bundleRequestExecutor.post(callback, emBundlerUrl + emBundlerStitchUri);
+
+        final AsylumCase responseData = response.getData();
+        Optional<List<IdValue<Bundle>>> maybeCaseBundles  = responseData.read(AsylumCaseFieldDefinition.CASE_BUNDLES);
+
+        final List<Bundle> caseBundles = maybeCaseBundles
+            .orElseThrow(() -> new IllegalStateException("caseBundle is not present"))
+            .stream()
+            .map(IdValue::getValue)
+            .collect(Collectors.toList());
+
+        if (caseBundles.size() != 1) {
+            throw new IllegalStateException("case bundles size is not 1 and is : " + caseBundles.size());
+        }
+
+        //stictchStatusflags -  NEW, IN_PROGRESS, DONE, FAILED
+        final String stitchStatus = caseBundles.get(0).getStitchStatus().orElse("");
```

Comment:
same as it was in notifications-api, we need to figure it out later for more than one bundle usage

Reply:
Agree Marcin. That would also depend on the UX design whether we plan to put alll bundles together or in separate collections. We can handle it later I think.